lms/djangoapps/ccx/tests/test_views.py

```diff
@@ -606,6 +647,44 @@ def test_manage_remove_single_student(self):
         self.assertIn(302, response.redirect_chain[0])
         self.assertEqual(outbox, [])
 
+    def test_manage_student_enrollment_limit(self):
+        """
+        Enroll students up to the enrollment limit
+        """
+        self.make_coach()
+        ccx = self.make_ccx(max_students_allowed=1)
+        ccx_course_key = CCXLocator.from_course_locator(self.course.id, ccx.id)
+        students = [
+            UserFactory.create(username="test{0}".format(seq), is_staff=False) for seq in range(2)
+        ]
+        url = reverse(
+            'ccx_manage_student',
+            kwargs={'course_id': CCXLocator.from_course_locator(self.course.id, ccx.id)}
+        )
+        # enroll the first student
+        data = {
+            'student-action': 'add',
+            'student-id': u','.join([students[0].email, ]),  # pylint: disable=no-member
+        }
+        response = self.client.post(url, data=data, follow=True)
+        self.assertEqual(response.status_code, 200)
+        # a CcxMembership exists for this student
+        self.assertTrue(
+            CourseEnrollment.objects.filter(course_id=ccx_course_key, user=students[0]).exists()
+        )
+        # try to enroll the second student without success
+        # enroll the first student
+        data = {
```

Comment:
However, this test and `test_enroll_single_non_user_student` are really similar. You can probably at minimum refactor these tests to use helper functions. And keep DDT in mind - we really want to prevent writing the same code in multiple places.